lib/datadog/core/telemetry/client.rb

```diff
@@ -37,21 +42,24 @@ def disable!
         def started!
           return if !@enabled || forked?
 
-          res = @emitter.request(:'app-started')
+          res = @emitter.request(Event::AppStarted.new)
```

Comment:
I like the use of a class. Nit, but does it make sense to initialize an object?

Reply:
This pattern of initializing an object makes sense for the telemetry events like `AppClientConfigurationChange`: https://github.com/DataDog/dd-trace-rb/blob/fb2ca4dc49d813f0a61e27534d7d72dfd76e5088/lib/datadog/core/telemetry/client.rb#L82

`AppClientConfigurationChange` needs information to be provided at two different times: first he needs to know what has changed in the source of that change (`changes, 'remote_config'`), then later he needs to receive the auto-incrementing `seq_id`: https://github.com/DataDog/dd-trace-rb/blob/fb2ca4dc49d813f0a61e27534d7d72dfd76e5088/lib/datadog/core/telemetry/request.rb#L20

Because these two sets of information exist in different context, I need a way to start a first set (`changes, 'remote_config'`) and then merge it with the second set (`seq_id`).
So this case it makes sense to initialize the class, then call amended accepting a parameter.

Most events are simple, like you mention `AppStarted`, and don't actually need two sets of data to build the event, but in order to keep the API consistent, I had to add this flexibility to all events.

Let me know if this makes sense or if you have further suggestions.

Reply:
I see: sometimes it has state attached. I suppose most of these events don't fire often, but if they ever fired as often as traces (crazy possibly, I know), then I was imagining this would apply more pressure to memory.

Still I do like the `event` as a parameter from a flexibility point of view.

Feedback is not blocking.